### PR TITLE
docs(guides website): Add vector tap guide

### PR DESCRIPTION
**Pull request by 001wwang**
Ref #8292 

This PR adds a guide to introduce users to the basics of `vector tap`. It includes an explanation of core features and a simple example use case.

[Rendered](https://github.com/vectordotdev/vector/blob/vector-tap-guide/website/content/en/guides/level-up/vector-tap-guide.md)

### CODE REVIEW

**Review by leebenson — CHANGES_REQUESTED** [review]
Could you fix the Netlify CI preview step please? I'd like to read this as it'll appear on the site. Thanks.

**Comment by 001wwang** [comment]
> Could you fix the Netlify CI preview step please? I'd like to read this as it'll appear on the site. Thanks.

Yep, here it is: https://deploy-preview-11517--vector-project.netlify.app/guides/level-up/vector-tap-guide/

File: website/content/en/guides/level-up/vector-tap-guide.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+
+[sinks.out]
+type = "blackhole"
+inputs = ["in*"]
+```
+
+Run Vector with this configuration and watch the configuration for changes.
+
+```console
+vector --config {config} -w
```
Comment (leebenson) [inline website/content/en/guides/level-up/vector-tap-guide.md:53]: Minor nit:

```suggestion
vector --config path/to/config.toml -w
```

This is likely obvious to the majority of users, but I could imagine someone being confused by seeing `{config}` and not realising that we're asking them to put the config snippet into a _file_ and reference it by path.

File: website/content/en/guides/level-up/vector-tap-guide.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+```
+
+Now run `vector tap`! You should see start seeing a stream of
+notifications (sent to `stderr`) and events (sent to `stdout`) in your terminal.
```
Comment (leebenson) [inline website/content/en/guides/level-up/vector-tap-guide.md:57]: Not directly related, but this caught my attention -- should _matched_ patterns also be sent to `stderr`, even though they're technically 'good' notifications?

Comment (001wwang) [reply website/content/en/guides/level-up/vector-tap-guide.md:57]: Yep, good eye. Currently, all notifications even "good" ones are sent to `stderr`.

**Review by leebenson — APPROVED** [review]
This is an excellent write up, solid work @001wwang. I can see users getting a lot of value out of this walk-through.

One attribute of tap that isn't discussed, but may be worth mentioning in the context of debugging, is the ability to tap _remote_ Vector instances -- at least those that are reachable over the network.

I think that's a stand out feature that's worth highlighting. My expectation running `vector tap` without any context, would be that it'd be doing some kind of local process communication, and that `vector tap` would need to be called from the same binary that spawned the agent.

The fact it can work over a network, I'd find quite impressive as a user. I imagine that could be a common use case in lieu of SSH'ing into a box.

In any case, I think the tone here is just right. Awesome 👍🏻

File: website/content/en/guides/level-up/vector-tap-guide.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+output patterns as additional arguments.
+
+Running the bare `vector tap` command invokes sensible defaults and is
+equivalent to specifying `--outputs-of "*"` where `"*"` matches any component
+ID.
+
+Try running `vector tap --inputs-of "out"`. The events should be the same
+as before, but you're now tapping the input events of the `blackhole` sink
+`out`. Note that tapping a component's input is effectively a shorthand for
+tapping all the outputs that feed into that component.
```
Comment (spencergilbert) [inline website/content/en/guides/level-up/vector-tap-guide.md:88]: I feel like this note should get different formatting... 🤔

**Review by spencergilbert — APPROVED** [review]
LGTM

Sidenote: my understanding is that we'll be moving to YAML for "default" configuration, I wonder when/if we want to start transitioning guides

File: website/content/en/guides/level-up/vector-tap-guide.md (hunk 3 above)
Comment (001wwang) [reply website/content/en/guides/level-up/vector-tap-guide.md:88]: Sure thing, I agree it's worth calling out more obviously.

**Comment by 001wwang** [comment]
> LGTM
> 
> Sidenote: my understanding is that we'll be moving to YAML for "default" configuration, I wonder when/if we want to start transitioning guides

Great point, I forgot about that. I'm happy to change the configs here to YAML to get ahead of it. Although, I wonder if we could eventually add a `JSON`, `YAML`, `TOML` toggle in the code blocks to switch between formats. cc: @jszwedko for more thoughts here.

**Comment by 001wwang** [comment]
> The fact it can work over a network, I'd find quite impressive as a user. I imagine that could be a common use case in lieu of SSH'ing into a box.

Sounds good, I added an info block in the `Under the Hood` section calling this out.

**Comment by jszwedko** [comment]
> > LGTM
> > Sidenote: my understanding is that we'll be moving to YAML for "default" configuration, I wonder when/if we want to start transitioning guides
> 
> Great point, I forgot about that. I'm happy to change the configs here to YAML to get ahead of it. Although, I wonder if we could eventually add a `JSON`, `YAML`, `TOML` toggle in the code blocks to switch between formats. cc: @jszwedko for more thoughts here.

I'd just stick with TOML for now since that is the language that the vast majority of examples are in. If/when we standardize on YAML we can switch everything over to that. Ideally, yes, it'd be nice to have all of the examples have toggles, but that doesn't need to block this PR since it likely requires some UI work.

File: website/content/en/guides/level-up/vector-tap-guide.md [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+By default, `tap` outputs events encoded in JSON format. YAML and logfmt are
+also supported and can be switched to using the `--format` option.
+
+Events in `tap` are actually sampled from their tapped components for
+performance and reliability. You can change the time interval of each sample
+using `--interval` and the maximum number of events to sample per interval with
+`--limit`.
+
+Try running `vector tap --quiet --format logfmt`. You'll now see no
+notifications and logfmt events.
```
Comment (jszwedko) [inline website/content/en/guides/level-up/vector-tap-guide.md:109]:
```suggestion
notifications and events encoded as logfmt.
```

I think this is a bit less ambiguous.

**Review by jszwedko — APPROVED** [review]
Nice work, this is a great write-up and I'm excited for it to drive more users to `vector tap`.

We might consider linking it in the existing troubleshooting guide since it is a great tool for that: https://vector.dev/guides/level-up/troubleshooting/

**Pull request merged**